cdap-docs/developers-manual/source/advanced/class-loading.rst

```diff
@@ -0,0 +1,140 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _class-loading:
+
+=============================================================
+Java Class Loading and Distributed Data Processing Frameworks
+=============================================================
+
+Java class loading is one of the most fundamental and powerful concepts provided by the
+Java Platform. Understanding the class loading mechanism helps you when designing and
+building extensible application frameworks. You can also avoid spending many hours in
+debugging exceptions such as ``ClassCastException`` and ``ClassNotFoundException``, among
+others. In this advanced section, we talk about how we used Java class loading to
+design an extensible application framework with a flexible class loading strategy.
+
+In CDAP, we provide a platform with which application developers are free to write big
+data applications without worrying about the underlying execution framework. In terms of
+class loading strategy, we provide these properties for an application:
+
+- Application can choose to use any library of any version;
+- Application can define a custom plugin API for extending itself;
+- Plugins can be written for an application by implementing the application plugin API; and
+- Different plugins can choose to use any library of any version, yet all plugins are
+  usable by the same application at the same time.
+
+
+Parent Delegation Model
```

Comment:
Should we move this to a a separate section and just add a reference to it? I feel documentation of Classloading in CDAP should have more emphasis on this page, and it starts from the next section onwards.